app/controllers/foreman_tasks/api/tasks_controller.rb

```diff
@@ -141,6 +141,33 @@ def bulk_cancel
         }
       end
 
+      api :POST, '/tasks/bulk_stop', N_('Stop all stoppable tasks')
+      param :search, String, :desc => N_('Stop tasks matching search string')
+      param :task_ids, Array, :desc => N_('Stop specific tasks by ID')
+      def bulk_stop
+        if params[:search].nil? && params[:task_ids].nil?
+          raise BadRequest, _('Please provide at least one of search or task_ids parameters in the request')
+        end
+
+        filtered_scope = bulk_scope
+        skipped, to_stop = filtered_scope.partition { |task| task.state == "stopped" }
```

Comment:
I know we did it this way bulk cancel, but the situation there was a bit different. In here we can tell if a task can be "force cancelled" or not before we load it. That means we can be more efficient and load only the records we will actually use and just count the others.

Reply:
Changed to 'where', thanks